pkg/devicerepository/config.go

```diff
@@ -0,0 +1,41 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package devicerepository
+
+import (
+	"context"
+
+	"go.thethings.network/lorawan-stack/v3/pkg/devicerepository/store"
+)
+
+// Config represents the DeviceRepository configuration.
+type Config struct {
+	Store store.Store `name:"-"`
+
+	Static    map[string][]byte `name:"-"`
+	Directory string            `name:"directory" description:"Retrieve devices from the filesystem"`
+	URL       string            `name:"url" description:"Retrieve devices from a web server"`
```

Comment:
Make the source explicit with a config key.

The implicit way of figuring out which source to use based on an order of precedence is legacy.